core/state_processor.go

```diff
@@ -338,25 +332,32 @@ func (p *StateProcessor) Process(block *types.WorkObject, batch ethdb.Batch) (ty
 			if _, ok := senders[tx.Hash()]; ok {
 				checkSig = false
 			}
-			fees, etxs, err, timing := ProcessQiTx(tx, p.hc, checkSig, header, batch, p.hc.headerDb, gp, usedGas, p.hc.pool.signer, p.hc.NodeLocation(), *p.config.ChainID, &etxRLimit, &etxPLimit, utxosCreatedDeleted)
+			qiTxFee, qiTxGas, fees, etxs, err, timing := ProcessQiTx(tx, p.hc, checkSig, firstQiTx, header, batch, p.hc.headerDb, gp, usedGas, p.hc.pool.signer, p.hc.NodeLocation(), *p.config.ChainID, &etxRLimit, &etxPLimit, utxosCreatedDeleted)
 			if err != nil {
 				return nil, nil, nil, nil, 0, 0, nil, fmt.Errorf("could not apply tx %d [%v]: %w", i, tx.Hash().Hex(), err)
 			}
+			firstQiTx = false
 			startEtxAppend := time.Now()
 			for _, etx := range etxs {
 				emittedEtxs = append(emittedEtxs, types.NewTx(etx))
 			}
 			totalEtxAppendTime += time.Since(startEtxAppend)
 			startEtxCoinbase := time.Now()
-			if block.Coinbase().IsInQiLedgerScope() {
-				totalFees.Add(totalFees, fees)
+
+			qiFees.Add(qiFees, fees)
+
+			// convert the fee to quai
+			qiTxFeeInQuai := misc.QiToQuai(primeTerminus.WorkObjectHeader(), qiTxFee)
+			// get the gas price by dividing the fee by qiTxGas
+			qiGasPrice := new(big.Int).Div(qiTxFeeInQuai, big.NewInt(int64(qiTxGas)))
+			if i == 0 {
+				minGasPrice = qiGasPrice
 			} else {
-				primeTerminus := p.hc.GetHeaderByHash(header.PrimeTerminusHash())
-				if primeTerminus == nil {
-					return nil, nil, nil, nil, 0, 0, nil, fmt.Errorf("could not find prime terminus header %032x", header.PrimeTerminusHash())
+				if minGasPrice.Cmp(qiGasPrice) < 0 {
```

Comment:
Should be > 0

Reply:
it should be less right? we want to find the min gas price among all txs, so checking if the existing min is less that the gas and setting it is the correct way i think

Reply:
you are right, changed this